service/history/replication/executable_sync_hsm_task.go

```diff
@@ -0,0 +1,241 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package replication
+
+import (
+	"context"
+	"time"
+
+	"go.temporal.io/api/serviceerror"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	replicationspb "go.temporal.io/server/api/replication/v1"
+	"go.temporal.io/server/common/definition"
+	"go.temporal.io/server/common/headers"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/namespace"
+	serviceerrors "go.temporal.io/server/common/serviceerror"
+	ctasks "go.temporal.io/server/common/tasks"
+	"go.temporal.io/server/service/history/shard"
+)
+
+// This is mostly copied from ExecutableActivityStateTask
+// The 4 replication executable task implemenatations are quite similar
+// we may want to do some refactoring later.
+
+type (
+	ExecutableSyncHSMTask struct {
+		ProcessToolBox
+
+		definition.WorkflowKey
+		ExecutableTask
+
+		taskAttr *replicationspb.SyncHSMAttributes
+
+		markPoisonPillAttempts int
+	}
+)
+
+var _ ctasks.Task = (*ExecutableSyncHSMTask)(nil)
+var _ TrackableExecutableTask = (*ExecutableSyncHSMTask)(nil)
+
+// var _ BatchableTask = (*ExecutableSyncHSMTask)(nil)
```

Comment:
remove this?

Reply:
I'd like this line to be uncommented if we ever implements batching for this task.